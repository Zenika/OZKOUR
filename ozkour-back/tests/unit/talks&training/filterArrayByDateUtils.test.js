const date = require('../../../src/utils/filterArrayByDateUtils')

const talks = [
  [
    {
      location: 'Singapour',
      eventType: 'Meetup',
      eventName: 'GraalVM Night',
      universe: 'web',
      date: '19/01/2021',
      speakers: 'Anne Imal',
      talkTitle: 'GraalVM for Sustainable Software Development?',
      link: 'https://www.meetup.com/singajug/events/275681145/',
      indexLine: 10
    },
    {
      location: 'Grenoble',
      eventType: 'NightClazz',
      eventName: 'NightClass',
      universe: 'web',
      date: '19/01/2021',
      speakers: 'Yves Rogne',
      talkTitle: 'Migration JS vers TS sur du react',
      link: 'https://www.meetup.com/singajug/events/275681145/',
      indexLine: 11
    },
    {
      location: 'Nantes',
      eventType: 'Meetup',
      eventName: 'Nantes JS #55',
      universe: 'web',
      date: '21/01/2021',
      speakers: 'Guy Tarenbois',
      talkTitle: 'Nuxt 2021',
      link: 'https://twitter.com/NantesJS/status/1351104198436392964',
      indexLine: 12
    },
    {
      location: 'Mix',
      eventType: 'Autre',
      eventName: 'Webinar Strigo',
      universe: 'web',
      date: '21/01/2021',
      speakers: 'Sarah Pelle',
      talkTitle: 'Simplify Remote Hands-On Training and Improve Engagement',
      link: 'https://zoom.us/webinar/register',
      indexLine: 13
    },
    {
      location: 'Nantes',
      eventType: 'NightClazz',
      eventName: 'RemoteClazz Nodejs',
      universe: 'web',
      date: '25/01/2021',
      speakers: 'Sophie Stiqué',
      talkTitle: 'Techniques minimalistes pour Node.js',
      link: 'https://www.meetup.com/NightClazz-by-Zenika-Nantes/events/275720340/',
      indexLine: 14
    },
    {
      location: 'Rennes',
      eventType: 'Meetup',
      eventName: 'DejTech Green IT',
      universe: 'web',
      date: '26/01/2021',
      speakers: 'Alin Verse',
      talkTitle: 'Introduction au numérique responsable',
      link: 'https://www.meetup.com/fr-FR/dejtech/events/275866319',
      indexLine: 14
    }
  ]
]

const talksWithUndefinedValue = [
  [
    {
      location: 'Singapour',
      eventType: 'Meetup',
      eventName: 'GraalVM Night',
      universe: 'web',
      date: '19/01/2021',
      speakers: 'Anne Imal',
      talkTitle: 'GraalVM for Sustainable Software Development?',
      link: 'https://www.meetup.com/singajug/events/275681145/',
      indexLine: 10
    },
    {
      location: 'Grenoble',
      eventType: 'NightClazz',
      eventName: 'NightClass',
      universe: 'web',
      date: '19/01/2021',
      speakers: 'Yves Rogne',
      talkTitle: 'Migration JS vers TS sur du react',
      link: 'https://www.meetup.com/singajug/events/275681145/',
      indexLine: 11
    },
    {
      location: 'Nantes',
      eventType: 'Meetup',
      eventName: 'Nantes JS #55',
      universe: '',
      date: '21/01/2021',
      speakers: 'Guy Tarenbois',
      talkTitle: 'Nuxt 2021',
      link: 'https://twitter.com/NantesJS/status/1351104198436392964',
      indexLine: 12
    },
    {
      location: 'Mix',
      eventType: 'Autre',
      eventName: 'Webinar Strigo',
      universe: '',
      date: '21/01/2021',
      speakers: 'Sarah Pelle',
      talkTitle: 'Simplify Remote Hands-On Training and Improve Engagement',
      link: 'https://zoom.us/webinar/register',
      indexLine: 13
    },
    {
      location: 'Nantes',
      eventType: 'NightClazz',
      eventName: 'RemoteClazz Nodejs',
      universe: '',
      date: '25/01/2021',
      speakers: 'Sophie Stiqué',
      talkTitle: 'Techniques minimalistes pour Node.js',
      link: 'https://www.meetup.com/NightClazz-by-Zenika-Nantes/events/275720340/',
      indexLine: 14
    },
    {
      location: 'Rennes',
      eventType: 'Meetup',
      eventName: 'DejTech Green IT',
      universe: 'web',
      date: '26/01/2021',
      speakers: 'Alin Verse',
      talkTitle: 'Introduction au numérique responsable',
      link: 'https://www.meetup.com/fr-FR/dejtech/events/275866319',
      indexLine: 14
    }
  ],
  [14, 13, 12]
]

describe('date filter', () => {
  it('should return an empty array', () => {
    const result = date.dateFilter()
    expect(result).toHaveLength(0)
  })
  it('should return only talk between 19-01-2021 and 25-012021', () => {
    const result = date.dateFilter(talks, '19-01-2021', '25-01-2021')
    expect(result).toMatchSnapshot()
  })
  it('should return an array with a length of 2', () => {
    const result = date.dateFilter(
      talksWithUndefinedValue,
      '19-01-2021',
      '25-01-2021'
    )
    expect(result.length).toBe(2)
  })
})
