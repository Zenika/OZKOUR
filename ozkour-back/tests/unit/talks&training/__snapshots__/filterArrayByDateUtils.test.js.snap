// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date filter should return only talk between 19-01-2021 and 25-012021 1`] = `
Array [
  Object {
    "date": "19/01/2021",
    "eventName": "GraalVM Night",
    "eventType": "Meetup",
    "indexLine": 10,
    "link": "https://www.meetup.com/singajug/events/275681145/",
    "location": "Singapour",
    "speakers": "Anne Imal",
    "talkTitle": "GraalVM for Sustainable Software Development?",
    "universe": "web",
  },
  Object {
    "date": "19/01/2021",
    "eventName": "NightClass",
    "eventType": "NightClazz",
    "indexLine": 11,
    "link": "https://www.meetup.com/singajug/events/275681145/",
    "location": "Grenoble",
    "speakers": "Yves Rogne",
    "talkTitle": "Migration JS vers TS sur du react",
    "universe": "web",
  },
  Object {
    "date": "21/01/2021",
    "eventName": "Nantes JS #55",
    "eventType": "Meetup",
    "indexLine": 12,
    "link": "https://twitter.com/NantesJS/status/1351104198436392964",
    "location": "Nantes",
    "speakers": "Guy Tarenbois",
    "talkTitle": "Nuxt 2021",
    "universe": "web",
  },
  Object {
    "date": "21/01/2021",
    "eventName": "Webinar Strigo",
    "eventType": "Autre",
    "indexLine": 13,
    "link": "https://zoom.us/webinar/register",
    "location": "Mix",
    "speakers": "Sarah Pelle",
    "talkTitle": "Simplify Remote Hands-On Training and Improve Engagement",
    "universe": "web",
  },
  Object {
    "date": "25/01/2021",
    "eventName": "RemoteClazz Nodejs",
    "eventType": "NightClazz",
    "indexLine": 14,
    "link": "https://www.meetup.com/NightClazz-by-Zenika-Nantes/events/275720340/",
    "location": "Nantes",
    "speakers": "Sophie Stiqu√©",
    "talkTitle": "Techniques minimalistes pour Node.js",
    "universe": "web",
  },
]
`;
